gemma-core/src/main/java/ubic/gemma/persistence/service/common/description/ExternalDatabaseService.java

```diff
@@ -19,22 +19,63 @@
 package ubic.gemma.persistence.service.common.description;
 
 import org.springframework.security.access.annotation.Secured;
+import ubic.gemma.model.common.auditAndSecurity.User;
 import ubic.gemma.model.common.description.ExternalDatabase;
 import ubic.gemma.persistence.service.BaseService;
 
+import javax.annotation.Nullable;
+import java.net.URL;
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+
 /**
  * @author Gemma
  */
 public interface ExternalDatabaseService extends BaseService<ExternalDatabase> {
 
+    @Override
+    @Secured({ "GROUP_ADMIN" })
+    Collection<ExternalDatabase> create( Collection<ExternalDatabase> entities );
+
+    @Override
+    @Secured({ "GROUP_ADMIN" })
+    ExternalDatabase create( ExternalDatabase entity );
+
     ExternalDatabase findByName( String name );
 
+    @Secured({ "GROUP_ADMIN", "GROUP_AGENT" })
+    ExternalDatabase findByNameWithAuditTrail( String name );
+
     @Override
-    @Secured({ "GROUP_USER" })
+    @Secured({ "GROUP_ADMIN" })
```

Comment:
We should probably remove external database creation from the Persister logic.

Reply:
It's fine, the persister uses the DAO, so it is not affected by this more stringent permission.